libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1740,6 +1741,15 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 					resultType->toString() +
 					"\"."
 				);
+
+			auto argArrayType = dynamic_cast<ArrayType const*>(argType.get());
+			auto resultArrayType = dynamic_cast<ArrayType const*>(resultType.get());
+			solAssert(
+				!argArrayType || argArrayType->location() != DataLocation::Storage ||
+				((resultArrayType->isPointer() || (argArrayType->isByteArray() && resultArrayType->isByteArray())) &&
```

Comment:
I think ``resultArrayType`` could be ``nullptr`` here.

Reply:
This assertion should probably only happen if the type *are* explicitly convertible (note that ``m_errorReporter.typeErrror`` does not throw).

For example, the following will crash this way:

```
contract C {
	uint[] a;
	function f() public pure {
		uint[] storage b = a;
		uint c = uint(uint[](b));
	}
}
```

Reply:
Yeah, this should be fixed

Reply:
fixed and added one test case.

Reply:
You mean in ``CompilerUtils.cpp``? Yes, that part isn't run, if there are errors in the type checker - so we have to gracefully create a error here in the TypeChecker, so that we can later assert in convertType :-). But yes, the latest change will take care of that!